### PR TITLE
$<-.grouped_df protects grouping variables

### PR DESCRIPTION
**Pull request by romainfrancois**
#4004 

``` r
library(dplyr)

d <- tibble(
  num = c(  1,   1,   2,   2,   3,   3),
  chr = c("a", "b", "c", "a", "b", "c"),
  val = 1:6
)

g <- group_by(d, num, chr)
g$num <- 3
#> Error: Column `num` can't be modified because it's a grouping variable
g$x <- 4
g
#> # A tibble: 6 x 4
#> # Groups:   num, chr [6]
#>     num chr     val     x
#>   <dbl> <chr> <int> <dbl>
#> 1     1 a         1     4
#> 2     1 b         2     4
#> 3     2 c         3     4
#> 4     2 a         4     4
#> 5     3 b         5     4
#> 6     3 c         6     4
```

<sup>Created on 2019-01-29 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1.9000)</sup>

### CODE REVIEW

**Comment by romainfrancois** [comment]
closes #4004

File: R/grouped-df.r [hunk 1]
```diff
@@ -186,6 +186,15 @@ ungroup.grouped_df <- function(x, ...) {
   }
 }
 
+#' @export
```
Comment (hadley) [inline R/grouped-df.r]: If you're going to implement this, I think you should also implement `[[<-`, and `[<-`. That's probably worth waiting for 0.9.0 so we can think it through carefully.

Comment (romainfrancois) [reply R/grouped-df.r]: 😬 yeah ok, those are usually more weird/work.

**Comment by hadley** [comment]
Closing because I think we're going to switch to regrouping consistently everywhere.

**Pull request closed without merging**